src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -350,27 +349,38 @@ public OpenIdConnectHandler(HttpClient backchannel, HtmlEncoder htmlEncoder)
 
             try
             {
-                var messageReceivedContext = await RunMessageReceivedEventAsync(authorizationResponse);
-                if (CheckEventResult(messageReceivedContext, out result))
+                AuthenticationProperties properties = null;
+                if (!string.IsNullOrEmpty(authorizationResponse.State))
+                {
+                    properties = Options.StateDataFormat.Unprotect(Uri.UnescapeDataString(authorizationResponse.State));
+                }
+
+                var messageReceivedContext = await RunMessageReceivedEventAsync(authorizationResponse, properties);
```

Comment:
Curious, what's the concrete use case behind this change? Why do we need to restore the properties bag before calling the event?